tests/main/interfaces-browser-support/task.yaml

```diff
@@ -0,0 +1,138 @@
+    snapbuild browser-support-consumer browser-support-consumer
+    snap install --dangerous browser-support-consumer/*.snap
+    rm -rf browser-support-consumer
+
+restore: |
+    rm -f *.error /var/tmp/test
+    for file in $OWNED_FILES $READABLE_FILES $READABLE_WITH_SANDBOX_FILES; do
+        rm -f $file
+    done
+
+    for dir in $(cat created_dirs); do
+        rm -rf $dir
+    done
+    rm -f created_dirs
+
+execute: |
+    . "$TESTSLIB/dirs.sh"
+
+    CONNECTED_PATTERN=":browser-support +browser-support-consumer"
+    DISCONNECTED_PATTERN="^\- +browser-support-consumer:browser-support"
+
+    echo "Then the plug is connected by default"
+    snap interfaces | MATCH "$CONNECTED_PATTERN"
+
+    echo "And the snap is able to access tmp"
+    echo "test" > /var/tmp/test
+    su -l -c "$SNAPMOUNTDIR/bin/browser-support-consumer.cmd ls /var/tmp/" test | MATCH test
+
+    echo "And the snap is able to access owned files"
+    for owned_file in $OWNED_FILES; do
+        echo "test" > "$owned_file"
+        chown test:12345 "$owned_file"
+        su -l -c "$SNAPMOUNTDIR/bin/browser-support-consumer.cmd cat $owned_file" test | MATCH test
+        su -l -c "$SNAPMOUNTDIR/bin/browser-support-consumer.cmd touch $owned_file" test
+    done
+
+    echo "And the snap is able to access readable files"
+    for readable_file in $READABLE_FILES; do
+        parent_dir=$(dirname $readable_file)
+        if [ ! -d $parent_dir ]; then
+            if mkdir -p $parent_dir; then
+                echo "$parent_dir" >> created_dirs
+            else
+                echo "$parent_dir couldn't be created, write-only partition?"
+                continue
+            fi
+        fi
+        echo "test" > "$readable_file"
+        su -l -c "$SNAPMOUNTDIR/bin/browser-support-consumer.cmd cat $readable_file" test | MATCH test
+    done
+
+    if [ "$ALLOW_SANDBOX" = "true" ]; then
+        for readable_file in $READABLE_WITH_SANDBOX_FILES; do
+            parent_dir=$(dirname $readable_file)
+            if [ ! -d $parent_dir ]; then
+                mkdir -p $parent_dir
+                echo "$parent_dir" >> created_dirs
+            fi
+            echo "test" > "$readable_file"
+            su -l -c "$SNAPMOUNTDIR/bin/browser-support-consumer.cmd cat $readable_file" test | MATCH test
+        done
+    fi
+
+    if [ "$(snap debug confinement)" = strict ] ; then
+        echo "When the plug is disconnected"
+        snap disconnect browser-support-consumer:browser-support
+        snap interfaces | MATCH "$DISCONNECTED_PATTERN"
+
+        echo "Then the snap is not able to access tmp"
+        if su -l -c "$SNAPMOUNTDIR/bin/browser-support-consumer.cmd ls /var/tmp/ 2>${PWD}/tmpdir-access.err" test; then
+            echo "Expected error with disconnected plug didn't happen"
+            exit 1
+        fi
+        cat tmpdir-access.err | MATCH "Permission denied"
```

Comment:
This shellcheck improvement apply for all the similar lines. The idea is to don't use cat and instead pipe directly the file to the match command like ==> MATCH "Permission denied" < tmpdir-access.err

More info at: https://github.com/koalaman/shellcheck/wiki/SC2002